sklearn/linear_model/_glm/glm.py

```diff
@@ -802,18 +807,20 @@ def __init__(
     @property
     def family(self):
         """Return the family of the regressor."""
-        # We use a property with a setter to make sure that the family is
-        # always a Tweedie distribution, and that self.power and
-        # self.family.power are identical by construction.
-        dist = TweedieDistribution(power=self.power)
-        # TODO: make the returned object immutable
-        return dist
+        # Make this attribute read-only to avoid mis-uses e.g. in GridSearch.
+        return "tweedie"
 
     @family.setter
     def family(self, value):
-        if isinstance(value, TweedieDistribution):
-            self.power = value.power
-        else:
-            raise TypeError(
-                "TweedieRegressor.family must be of type TweedieDistribution!"
-            )
+        if value != "tweedie":
+            raise ValueError("TweedieRegressor.family must be 'tweedie'!")
+
+    def _get_family_instance(self):
+        """Return an instance of ``TweedieDistribution``.
+
+        Returns
+        -------
+        family_instance : TweedieDistribution
+            Returns an instance of TweedieDistribution.
+        """
+        return TweedieDistribution(power=self.power)
```

Comment:
I think I would be in favor of removing the `family` property and instead validate the consistency of `self.family` in `_get_family_instance` which should be only called at `fit` time instead of `__init__` or `set_params` time:

```suggestion
      if isinstance(self.family, TweedieDistribution):
          # ignore self.power if the user passed a distribution instance directly
          # for backward compat
          return self.family
      elif self.family == "tweedie":
          return TweedieDistribution(power=self.power)
      else:
          raise ValueError(
              "TweedieRegressor.family must always be 'tweedie', "
              f"got: {self.family!r}"
          )
```

disclaimer: untested, there might be typos.